torch/utils/_sympy/value_ranges.py

```diff
@@ -500,6 +500,53 @@ def or_(a, b):
     def and_(a, b):
         return ValueRanges.coordinatewise_increasing_map(a, b, sympy.And)
 
+    @staticmethod
+    def _bool_to_int(x):
+        if x.is_singleton():
+            return ValueRanges.wrap(sympy.Integer(1 if x.lower else 0))
+        else:
+            return ValueRanges(sympy.Integer(0), sympy.Integer(1))
+
+    @staticmethod
+    def _bit_length(n):
+        return len(bin(n).lstrip("-0b"))
+
+    @classmethod
+    def bitwise_and(cls, a, b):
+        a, b = ValueRanges.wrap(a), ValueRanges.wrap(b)
+        if a.is_bool:
+            a = cls._bool_to_int(a)
+        if b.is_bool:
+            b = cls._bool_to_int(b)
```

Comment:
This is not right. The problem is that bitwise operations on booleans are defined to return boolean:

```
>>> True | False
True
```

so you are obligated to return a boolean ValueRanges when both arguments are boolean. We are also a little loosey-goosey about this, but my general preference is to insert explicit coercions when they would have happened, e.g., `2 | True` should become `2 | bool_to_int(True)`